core/chains/evm/logpoller/log_poller.go

```diff
@@ -497,7 +505,20 @@ func (lp *logPoller) pollAndSaveLogs(ctx context.Context, currentBlockNumber int
 			lp.lggr.Warnw("Unable to save logs resuming from last saved block + 1", "err", err, "block", currentBlockNumber)
 			return
 		}
+		// Update current block.
+		// Same reorg detection on unfinalized blocks.
 		currentBlockNumber++
+		if currentBlockNumber > latestBlockNumber {
```

Comment:
We can't make this the loop condition anymore? `for currentBlockNumber <= latestBlockNumber`?

Reply:
you'd still need the break since we want to increment before calling getCurrentBlockMaybeHandleReorg